### ISSUE TITLE
test failure in VariantRecord_test.py::test_ascii_annotation_with_variable_length_can_be_added

### ISSUE THREAD

**Issue by nieder** [pysam-developers/pysam#843]
When running tests for pysam-py-0.15.3, tests fail like this. All tests passed on 0.15.2. This is macOS 10.13.6 with python-2.7.16 and htslib-1.9
````
tests/VariantFile_test.py::TestUnicode::test_record_with_unicode PASSED
tests/VariantRecord_test.py::test_ascii_annotation_can_be_added PASSED
tests/VariantRecord_test.py::test_ascii_annotation_with_variable_length_can_be_added Assertion failed: (!fmt->p_free), function bcf_update_format, file vcf.c, line 3496.
/tmp/fink.7y4xf: line 6: 97317 Abort trap: 6           /sw/bin/pytest-2.7 tests
````

**Comment by jmarshall** [comment]
This is the same problem as #640, caused by samtools/htslib#813 which has been fixed in HTSlib — the fix is on HTSlib **develop** and will appear in its next release.

**Comment by jmarshall** [comment]
This has been fixed since pysam 0.16.0 (when used with HTSlib 1.10 — which fixed the underlying problem — or later).
